core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java

```diff
@@ -108,4 +128,14 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
     }
     return actualData;
   }
+
+  @Override public Object getDataBasedOnDataTypeFromNoDictionary(ByteBuffer data) {
```

Comment:
I don't think it is required to have one more method, just rename and handle in same method `getDataBasedOnDataTypeFromSurrogates` , it can avoid duplicate code and extra handling